code-challenges.test.js

```diff
@@ -15,37 +15,126 @@
+//Puesdo C:
+// Create a function called shuffle array to match the test.
+// Give the funciton one parameter 'array' too match the data to bee arguments.
+//Start the function by designating a new variable called remove first that will take out the first item from the array. 
+// Store indside the variable the parameter array and use the .shift() method to remove the first item.
+// Next, create a for loop that will iterate through the array and set the the index to equal the array.length - 1. 
+// Next, set i > 0 and we want i to decrement so set i --
+// inside the for loop create a new variable that will hold the logic for the Fisher-Yates algorithm. 
+// This will give a random index within the remaining elements of the array, and then swap the curren element with the random one. 
+// Last return the array that has removed the first value and shuffled the rest. 
+
+const shuffleArray = (array) => {
+  let removeFirst = array.shift()
+  for (let i = array.length - 1; i > 0; i--) {
+    let newArray = Math.floor(Math.random() * (i + 1))
+    let a = array[newArray]
+    let b = array[i]
+    array[i] = a
+    array[newArray] = b
+  }
+  return array
+}
+
+// PASS  ./code-challenges.test.js
+// shuffleArray
+//   ✓ returns a new array that has removed the first item from the original array and shuffled the rest of the items. (1 ms)
+
+// Test Suites: 1 passed, 1 total
+// Tests:       1 passed, 1 total
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the end tally.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = { upVotes: 13, downVotes: 2 }
-// Expected output: 11
-const votes2 = { upVotes: 2, downVotes: 33 }
-// Expected output: -31
+describe("totalVotes", () => {
+  const votes1 = { upVotes: 13, downVotes: 2 }
+  // Expected output: 11
+  const votes2 = { upVotes: 2, downVotes: 33 }
+  //Expected output: -31
+  it("Function that takes in an object containing up and down votes exectues a mathmatical expression to find the end total.", () => {
+    expect(totalVotes(votes1)).toEqual(11)
+    expect(totalVotes(votes2)).toEqual(-31)
+  })
+})
 
 // b) Create the function that makes the test pass.
+//Puesdo C:
+//Create the functions named above(totalVotes).
+// Give the funciton one parameter called 'object' to match the data we are passing in.
+// To find the number of votes create a simple mathmatical formula of upvote - downvote. This will give the total.
+// Since the data is in an object we need to call on the key-value pair to retrieve the information.
+// This can be accomplied by dot notation.
+// Place the mathmatical formula in a variable called total, so we can just return the variable for cleanliness.
+// Breathe because your crushing it...
+
+const totalVotes = (object) => {
+  let uVotes = object.upVotes
+  let dVotes = object.downVotes
+  let total = uVotes - dVotes
+  return total
+}
```

Comment:
This definitely works and your logic matches your pseudo code. As a challenge, think about how you could reduce the number of variables you have to create!